src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/AssociatedMetadataTypeTypeDescriptionProvider.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.ComponentModel.DataAnnotations
+{
+    /// <summary>
+    /// Extends the metadata information for a class by adding attributes and property
+    /// information that is defined in an associated class.
+    /// </summary>
+    public class AssociatedMetadataTypeTypeDescriptionProvider : TypeDescriptionProvider
+    {
+        private Type _associatedMetadataType;
+
+        /// <summary>
+        /// Initializes a new instance of the System.ComponentModel.DataAnnotations.AssociatedMetadataTypeTypeDescriptionProvider
+        /// class by using the specified type.
+        /// </summary>
+        /// <param name="type">The type for which the metadata provider is created.</param>
+        public AssociatedMetadataTypeTypeDescriptionProvider(Type type)
+            : base(TypeDescriptor.GetProvider(type))
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the System.ComponentModel.DataAnnotations.AssociatedMetadataTypeTypeDescriptionProvider
+        /// class by using the specified metadata provider type and associated type.
+        /// </summary>
+        /// <param name="type">The type for which the metadata provider is created.</param>
+        /// <param name="associatedMetadataType">The associated type that contains the metadata.</param>
+        /// <exception cref="System.ArgumentNullException">The value of associatedMetadataType is null.</exception>
+        public AssociatedMetadataTypeTypeDescriptionProvider(Type type, Type associatedMetadataType)
+            : this(type)
+        {
+            if (associatedMetadataType == null)
+            {
+                throw new ArgumentNullException(nameof(associatedMetadataType));
```

Comment:
@ajcvickers I addressed the [code review](https://github.com/dotnet/corefx/pull/33312#pullrequestreview-174629784)